### PR TITLE
Check workers placement with strict anti affinity

### PR DESCRIPTION
**Pull request by itzikb-redhat**
Check that all the worker have the same server group and are placed on different hosts in case of strict
anti affinity

### CODE REVIEW

File: test/extended/openstack/servergroup.go [hunk 1]
```diff
@@ -121,8 +134,60 @@ var _ = g.Describe("[sig-installer][Feature:openstack] The OpenStack platform",
...
+			serverGroupsWithThatName, err := serverGroupIDsFromName(computeClient, workerGroupName)
+			o.Expect(serverGroupsWithThatName, err).To(o.HaveLen(1), "the server group name either was not found or is not unique")
+
+			serverGroup, err := servergroups.Get(computeClient, serverGroupsWithThatName[0]).Extract()
+			o.Expect(serverGroup.Members, err).To(o.ContainElements(workerInstanceUUIDs...))
+		}
 
 	})
+
+	// OCP 4.10: https://issues.redhat.com/browse/OSASINFRA-2507
```
Comment (pierreprinetti) [inline test/extended/openstack/servergroup.go:172]: I think it's `OCP 4.10: https://issues.redhat.com/browse/OSASINFRA-2570`

File: test/extended/openstack/servergroup.go [hunk 2]
```diff
@@ -121,8 +134,60 @@ var _ = g.Describe("[sig-installer][Feature:openstack] The OpenStack platform",
...
+
+				instanceWorkerGroupNameField := getFromUnstructured(res, "spec", "providerSpec", "value", "serverGroupName")
+				o.Expect(instanceWorkerGroupNameField).NotTo(o.BeNil(), "the Server group name should be present in the Machine definition")
+
+				instanceWorkerGroupName := instanceWorkerGroupNameField.(string)
+				o.Expect(instanceWorkerGroupName).NotTo(o.BeEmpty(), "the Server group name should not be the empty string")
+				if workerGroupName == "" {
+					workerGroupName = instanceWorkerGroupName
+				} else {
+					o.Expect(instanceWorkerGroupName).To(o.Equal(workerGroupName), "two Control plane Machines have different controlPlaneGroupName set")
```
Comment (pierreprinetti) [inline test/extended/openstack/servergroup.go:157]: s/Control plane/worker/

**Comment by pierreprinetti** [comment]
LGTM! Just a couple nits.

File: test/extended/openstack/servergroup.go (hunk 1 above)
Comment (itzikb-redhat) [reply test/extended/openstack/servergroup.go:172]: Done

File: test/extended/openstack/servergroup.go (hunk 2 above)
Comment (itzikb-redhat) [reply test/extended/openstack/servergroup.go:157]: I modify the second to workerGroupName

**Pull request merged**